paddle/gserver/layers/WithFunction.cpp

```diff
@@ -0,0 +1,24 @@
+#include "WithFunction.h"
+#include "paddle/function/Function.h"
+namespace paddle {
+
+void WithFunction::appendFunction(std::vector<function::KernelType> *functions,
+                                  const std::string &name,
+                                  const FuncConfig &config,
+                                  bool useGPU) {
+  std::shared_ptr<FunctionBase> func;
```

Comment:
In https://github.com/PaddlePaddle/Paddle/issues/2050 which was claimed to be fixed with this PR, it says that we shoud **replace** `FunctionBase` with `std::function`.  But here it shows that `FunctionBase` is not replaced at all. We just wrap it by a `std::function`.  I believe that if a concept is not necessary, we should remove it, instead of **hide** it.